src/core/Directus/Database/TableGateway/RelationalTableGateway.php

```diff
@@ -608,7 +608,7 @@ public function addOrUpdateManyToOneRelationships($schema, $parentRow, &$childLo
             }
 
             // Update/Add foreign record
-            if ($this->recordDataContainsNonPrimaryKeyData($foreignRow, $foreignTableSchema->getPrimaryKeyName()) && ($field->getType() != "file" && !isset($parentRow[$fieldName][$primaryKey]))) {
+            if ($this->recordDataContainsNonPrimaryKeyData($foreignRow, $foreignTableSchema->getPrimaryKeyName()) && $field->getType() != "file" ) {
```

Comment:
Find a way to remove hardcoded condition like `$field->getType() != "file"`
Adding comment in TODO.